File: epi_judge_java/epi/strings/StringIntegerInterconversion.java
package epi.strings;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;
import epi.test_framework.TestFailure;

public class StringIntegerInterconversion {

    public static String intToString(int x) {
        if(x ==0) return "0";

        boolean isMinus = x < 0;
        StringBuilder builder = new StringBuilder();
        while (x != 0) {
            builder.append(Math.abs(x % 10));
            x /= 10;
        }

        if (isMinus) builder.append("-");
        builder.reverse();
        return builder.toString();
    }

    public static int stringToInt(String s) {
        int result = 0;
        boolean isMinus = s.charAt(0) == '-';
        for (int i = isNum(s.charAt(0)) ? 0 : 1; i < s.length(); i++) {
            char ch = s.charAt(i);
            result = result * 10 + (ch - '0');
        }
        if (isMinus) result = -result;
        return result;
    }

    private static boolean isNum(char ch) {
        return ch >= '0' && ch <= '9';
    }

    @EpiTest(testDataFile = "string_integer_interconversion.tsv")
    public static void wrapper(int x, String s) throws TestFailure {
        if (Integer.parseInt(intToString(x)) != x) {
            throw new TestFailure("Int to string conversion failed");
        }
        if (stringToInt(s) != x) {
            throw new TestFailure("String to int conversion failed");
        }
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "StringIntegerInterconversion.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_java/epi/strings/kt/StringIntegerInterconversion.kt
package epi.strings.kt

import epi.test_framework.EpiTest
import epi.test_framework.GenericTest
import epi.test_framework.TestFailure
import kotlin.math.abs

object StringIntegerInterconversion {

    @JvmStatic
    fun intToString(x: Int): String {
        if(x ==0) return "0";
        val isMinus = x < 0;
        var value = x;
        val builder = StringBuilder();
        while (value != 0) {
            builder.append(abs(value % 10));
            value /= 10;

        }

        if (isMinus) builder.append("-");
        builder.reverse();
        return builder.toString();
    }

    @JvmStatic
    fun stringToInt(s: String?): Int {
        val isMinus = s!![0] == '-';
        var result = 0;
        for (i in (if (Character.isDigit(s!![0])) 0 else 1) until s.length) {
            result = result * 10 + (s!![i] - '0');
        }
        if (isMinus) result = -result;
        return result;
    }

    @JvmStatic
    @EpiTest(testDataFile = "string_integer_interconversion.tsv")
    @Throws(TestFailure::class)
    fun wrapper(x: Int, s: String?) {
        if (intToString(x).toInt() != x) {
            throw TestFailure("Int to string conversion failed")
        }
        if (stringToInt(s) != x) {
            throw TestFailure("String to int conversion failed")
        }
    }

    @JvmStatic
    fun main(args: Array<String>) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "StringIntegerInterconversion.java",
                                object : Any() {}.javaClass.enclosingClass)
                        .ordinal)
    }
}